packages/block-library/src/query-loop/edit.js

```diff
@@ -82,17 +82,19 @@ export default function QueryLoopEdit( {
 
 			// When you insert this block outside of the edit site then store
 			// does not exist therefore we check for its existence.
+			// TODO: remove this code, edit-site shouldn't be called in block-library.
+			// This creates a cycle dependency.
```

Comment:
I'm not sure yet why the code here is needed but it may suggest a missing context or something like that. A block shouldn't be trying to detect its context without using the "context" API.

Reply:
Yes, totally agree, this is why I added the comment so it was clear that it needs further work to decouple the block's implementation from the context in which it is loaded. In this case, we need a way to pass down the information about the site context to the block.

Reply:
I'm working on this in a separate PR that will handle the preview of categories as well.